src/scripts/Build-ModuleDoc.ps1

```diff
@@ -0,0 +1,210 @@
+            Type         = $parameter.parameterValue
+        }
+    }
+
+    foreach($example in $help.examples.example)
+    {
+        $examples += @{
+            Id     = $example.title -replace '\D'
+            Remark = $example.remarks.text | Select-Object -First 1
+            Code   = $example.code
+        }
+    }
+
+    return @{
+        Description = $help.Description
+        Syntax      = ($help.syntax | Out-String) -replace '^[\r\n]+|\.|[\r\n]+$'
+        Synopsis    = $help.Synopsis
+        Examples    = $examples
+        Parameters  = $parameters
+    }
+}
+
+switch ($PSCmdlet.ParameterSetName)
+{
+    'ByPath'
+    {
+        if (-not [System.IO.Path]::IsPathRooted($Path))
+        {
+            $Path = Resolve-Path -Path $Path
+        }
+
+        Import-Module -FullyQualifiedName $Path -WarningAction 'SilentlyContinue'
+        $moduleName = [System.IO.Path]::GetFilenameWithoutExtension($Path)
+        $modulePath = Split-Path -Path $Path -Parent
+    }
+
+    'ByName'
+    {
+        if (-not (Get-Module -Name $Name -ListAvailable))
+        {
+            throw ('Module {0} not found.' -f $Name)
+        }
+
+        Import-Module -Name $Name -WarningAction 'SilentlyContinue'
+        $moduleName = $Name
+        $modulePath = Split-Path -Path (Get-Module -Name $moduleName).Path -Parent
+    }
+}
+
+$sb = [System.Text.StringBuilder]::new()
+[void]$sb.AppendLine('# {0} PowerShell Module' -f $moduleName).AppendLine()
+
+# Check for manifest, if it exists add Description and required modules
+$manifestPath = Get-ChildItem -Path (Split-Path -Path $modulePath -Parent) -Filter '*.psd1'
+if ($manifestPath)
+{
+    $manifest = Import-PowerShellDataFile -Path $manifestPath.FullName
+    [void]$sb.AppendLine($manifest.Description).Appendline()
+
+    if ($manifest.RequiredModules)
+    {
+        [void]$sb.AppendLine('## Required Modules').AppendLine()
+        [void]$sb.AppendLine('Name | Version')
+        [void]$sb.AppendLine('|:---:|:---:|')
+    
+        foreach ($requiredModule in $manifest.RequiredModules)
+        {
+            [void]$sb.AppendLine("$($requiredModule.ModuleName) | $($requiredModule.ModuleVersion)")
+        }
+
+        [void]$sb.AppendLine()
+    }        
+}
+
+# Parse functions
+[void]$sb.AppendLine('## Functions').AppendLine()
+$functions = Get-Command -Module $moduleName -CommandType 'Function'
+foreach ($function in $functions)
+{
+    $functionHelp = Get-HelpContent -Function $function.Name
+    [void]$sb.AppendLine("### $($function.Name)").AppendLine()
+
+    if (-not [String]::IsNullOrEmpty($functionHelp.syntax))
+    {
+        [void]$sb.AppendLine('#### Syntax').AppendLine()
+        [void]$sb.AppendLine("``$($functionHelp.Syntax)``").AppendLine()
+    }
+
+    if ($functionHelp.Parameters.Count -gt 0)
+    {
+        [void]$sb.AppendLine('#### Parameters').AppendLine()
+        [void]$sb.AppendLine('Name | Type | Description | Required? | Default Value')
```

Comment:
Should required be before the description? I feel like that's the typical order. Fwiw, all of our parameter docs put that first in the description.